### PR TITLE
fit-trackee: 0.7.31 -> 0.8.5

### PR DESCRIPTION
**Pull request by traxys**
This upgrade will require database migration updates

## Description of changes

Diff: https://github.com/SamR1/FitTrackee/compare/v0.7.31...v0.8.5
Changelog: https://github.com/SamR1/FitTrackee/blob/v0.8.5/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/geospatial/fit-trackee/default.nix [hunk 1]
```diff
@@ -25,29 +25,27 @@ let
...
 python.pkgs.buildPythonApplication rec {
   pname = "fit-trackee";
-  version = "0.7.31";
+  version = "0.8.5";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "SamR1";
     repo = "FitTrackee";
     rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/servers/geospatial/fit-trackee/default.nix:34]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/servers/geospatial/fit-trackee/default.nix [hunk 2]
```diff
@@ -25,29 +25,27 @@ let
...
     substituteInPlace pyproject.toml \
-      --replace-fail psycopg2-binary psycopg2 \
-      --replace-fail 'flask = "^3.0.2"' 'flask = "*"' \
-      --replace-fail 'pyopenssl = "^24.0.0"' 'pyopenssl = "*"' \
-      --replace-fail 'sqlalchemy = "=1.4.51"' 'sqlalchemy = "*"'
+      --replace-fail 'gunicorn = "^22.0.0"' 'gunicorn = "*"' \
+      --replace-fail psycopg2-binary psycopg2
   '';
 
   nativeBuildInputs = [
```
Comment (GaetanLepage) [inline pkgs/servers/geospatial/fit-trackee/default.nix:44]:
```suggestion
  build-system = [
```

File: pkgs/servers/geospatial/fit-trackee/default.nix [hunk 3]
```diff
@@ -25,29 +25,27 @@ let
 in
 python.pkgs.buildPythonApplication rec {
   pname = "fit-trackee";
-  version = "0.7.31";
+  version = "0.8.5";
   format = "pyproject";
```
Comment (GaetanLepage) [inline pkgs/servers/geospatial/fit-trackee/default.nix:29]:
```suggestion
  pyproject = true;
```

**Comment by traxys** [comment]
Result of `nixpkgs-review pr 328942` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 328942` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 328942` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 328942` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 328942` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 328942` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 328942` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by traxys** [comment]
Result of `nixpkgs-review pr 328942` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 328942` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 328942` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fit-trackee</li>
    <li>fit-trackee.dist</li>
  </ul>
</details>

**Pull request merged**